<launch>
  <arg name="policy_name"/>
  <arg name="rviz" default="true"/>
  <arg name="arm_id" default="panda"/>
  <arg name="robot_ip" default="franka_robot"/>
  <!-- we don't use the default gripper /-->
  <arg name="load_gripper" default="false" />
  <arg name="exp_name" default="exp_137"/>
  <arg name="ckpt_dirname" default="policy_checkpoint_0006400000"/>
  
  <rosparam file="$(find softgrasp_ros_controllers)/config/softgrasp_ros_controllers.yaml" subst_value="true"/>
  <rosparam file="$(find franka_example_controllers)/config/franka_example_controllers.yaml" />
  <include file="$(find softgrasp_ros_controllers)/launch/policy_actionserver.launch" ns="tf_policy_controller">
    <arg name="policy_name" value="$(arg policy_name)"/>
    <arg name="exp_dir" value="$(find softgrasp_ros_controllers)/policies/$(arg exp_name)"/>
    <arg name="ckpt_dirname" value="$(arg ckpt_dirname)"/>
  </include>
  <include file="$(find franka_control)/launch/franka_control.launch" >
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>
  <node name="tf_policy_controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="--stopped tf_policy_controller"/>
  <node name="impedance_controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="cartesian_impedance_example_controller"/>
  <node name="interactive_marker" pkg="franka_example_controllers" type="interactive_marker.py" required="true" output="screen">
    <param name="link_name" value="panda_link0" />
    <remap from="equilibrium_pose" to="cartesian_impedance_example_controller/equilibrium_pose" />
  </node> 
  <node pkg="rviz" type="rviz" output="screen" name="rviz" if="$(arg rviz)" args="-d $(find franka_example_controllers)/launch/rviz/franka_description_with_marker.rviz"/>
  
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false" />
  <node name="rqt_console" pkg="rqt_console" type="rqt_console" required="true"/>
  <include file="$(find softgrasp_ros_controllers)/launch/static_poses.launch"/>
</launch>